lib/hasher/include/SHA512Strategy.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef _SHA512_STRATEGY_HPP_
+#define _SHA512_STRATEGY_HPP_
+
+#include "HashStrategy.hpp"
+#include <string>
+#include <openssl/sha.h>
+
+namespace irods {
+    const std::string SHA512_NAME( "sha512" );
+    class SHA512Strategy : public HashStrategy {
+        public:
+            SHA512Strategy() {};
+            virtual ~SHA512Strategy() {};
+
+            virtual std::string name() const {
+                return SHA512_NAME;
+            }
+            virtual error init( boost::any& context ) const;
+            virtual error update( const std::string& data, boost::any& context ) const;
+            virtual error digest( std::string& messageDigest, boost::any& context ) const;
+            virtual bool isChecksum( const std::string& ) const;
```

Comment:
If all of these functions override a virtual function from the base class, you can remove the `virtual` keyword and append the `override` keyword to each one. For example:
```c++
error init(boost::any& context) const override;
```

Reply:
Done for all hashing schemes.